### PR TITLE
Re-bench

### LINKED ISSUE #1178: The `submit_finality_proof` exhaust block limits
```sh
2023-06-15 12:45:25 ERROR bin-s2s: [header-relay] [polkadot-to-crab] an error occurred for header relay Relay(Client(RPC("Rpc(ClientError(Call(Custom(ErrorObject { code: ServerError(1010), message: \"Invalid Transaction\", data: Some(RawValue(\"Transaction would exhaust the block limits\")) }))))")))
```

### PR DESCRIPTION
**Pull request by AurevoirXavier**
Closes #1178.

### CODE REVIEW

File: runtime/crab/src/pallets/bridge_grandpa.rs [hunk 1]
```diff
@@ -26,7 +26,7 @@ pub type PolkadotHeadersToKeep = ConstU32<500>;
 impl pallet_bridge_grandpa::Config<WithPolkadotGrandpa> for Runtime {
 	type BridgedChain = bp_darwinia::DarwiniaLike;
 	type HeadersToKeep = PolkadotHeadersToKeep;
-	type MaxBridgedAuthorities = ConstU32<100_000>;
+	type MaxBridgedAuthorities = ConstU32<4_096>;
```
Comment (AurevoirXavier) [inline runtime/crab/src/pallets/bridge_grandpa.rs:29]: @boundless-forest Please help check if this needs a migration. I'll consider revert this change.

**Review by boundless-forest — APPROVED** [review]

File: runtime/crab/src/weights/pallet_bridge_parachains.rs [hunk 2]
```diff
@@ -88,9 +96,9 @@ impl<T: frame_system::Config> pallet_bridge_parachains::WeightInfo for WeightInf
 		// Proof Size summary in bytes:
 		//  Measured:  `485`
 		//  Estimated: `77573`
-		// Minimum execution time: 84_508_000 picoseconds.
-		Weight::from_parts(84_508_000, 0)
-			.saturating_add(Weight::from_parts(0, 0))
+		// Minimum execution time: 29_118_000 picoseconds.
+		Weight::from_parts(29_990_000, 0)
+			.saturating_add(Weight::from_parts(0, 77573))
```
Comment (hackfisher) [inline runtime/crab/src/weights/pallet_bridge_parachains.rs:101]: Ref: https://github.com/darwinia-network/darwinia-messages-substrate/blob/main/modules/parachains/src/weights_ext.rs#L49

**Review by hackfisher — APPROVED** [review]

File: runtime/crab/src/weights/pallet_bridge_parachains.rs (hunk 2 above)
Comment (boundless-forest) [reply runtime/crab/src/weights/pallet_bridge_parachains.rs:101]:
```rs
Weight::from_parts(39_465_487_000, 0)
			.saturating_add(Weight::from_parts(0, 0))
			.saturating_add(T::DbWeight::get().reads(2050))
			.saturating_add(T::DbWeight::get().writes(3072))
```
The root cause of this call failure is the reads and writes times are so huge! this bloats the block limit.

File: runtime/crab/src/weights/pallet_bridge_parachains.rs [hunk 3]
```diff
@@ -57,22 +61,26 @@ impl<T: frame_system::Config> pallet_bridge_parachains::WeightInfo for WeightInf
...
+		//  Estimated: `72499 + p * (5074 ±0)`
+		// Minimum execution time: 23_588_000 picoseconds.
+		Weight::from_parts(24_031_000, 0)
+			.saturating_add(Weight::from_parts(0, 72499))
+			// Standard Error: 23_116
+			.saturating_add(Weight::from_parts(18_957_485, 0).saturating_mul(p.into()))
+			.saturating_add(T::DbWeight::get().reads(2))
+			.saturating_add(T::DbWeight::get().reads((2_u64).saturating_mul(p.into())))
+			.saturating_add(T::DbWeight::get().writes((3_u64).saturating_mul(p.into())))
+			.saturating_add(Weight::from_parts(0, 5074).saturating_mul(p.into()))
```
Comment (hackfisher) [inline runtime/crab/src/weights/pallet_bridge_parachains.rs:83]: What's the p here?

Comment (boundless-forest) [reply runtime/crab/src/weights/pallet_bridge_parachains.rs:83]: https://github.com/darwinia-network/darwinia-messages-substrate/blob/main/modules/parachains/src/weights_ext.rs#L52

It refers to `parachains_count`, it's normally a number less than 10.

**Pull request merged**